spacelift/resource_stack_dependency.go

```diff
@@ -0,0 +1,152 @@
+package spacelift
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/shurcooL/graphql"
+
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal"
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal/structs"
+)
+
+func resourceStackDependency() *schema.Resource {
+	return &schema.Resource{
+		Description: "`spacelift_stack_dependency` represents a Spacelift **stack dependency** - " +
+			"a dependency between two stacks. When one stack depends on another, the tracked runs " +
+			"of the stack will not start until the dependent stack is successfully finished.",
+
+		CreateContext: resourceStackDependencyCreate,
+		ReadContext:   resourceStackDependencyRead,
+		UpdateContext: resourceStackDependencyUpdate,
+		DeleteContext: resourceStackDependencyDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceStackDependencyImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"stack_id": {
+				Type:        schema.TypeString,
+				Description: "immutable ID (slug) of stack which has a dependency.",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"depends_on_stack_id": {
+				Type:        schema.TypeString,
+				Description: "immutable ID (slug) of stack to depend on.",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"triggers": {
+				Type:        schema.TypeBool,
+				Description: "describes whether we should trigger the dependent if it's not triggered by the push, but the current stack has changed. Defaults to `true`.",
```

Comment:
I'm not sold on this defaulting to true. Shouldn't this only be true in a very limited set of circumstances (after we have stack output references - only whenever a stack has an output that is passed out of band?)?

Reply:
Hmm, I think I understand what you mean. My feeling is that while we don't have output references, default behaviour set to true makes more sense, whereas when having output references a default behaviour of false will make more sense.

After we have stack output references, the behaviour will be like that:
- if there was a change in outputs that are referenced in current stack -> behaviour similar to triggers:true
- if there were no changes -> behaviour similar to triggers:false -> unless we specified differently

Generally the idea with this field was to give users option to control how prefilling should work. Maybe we should actually not be setting defaults here, and require explicit definition? 🤔

Reply:
We could make this mandatory without default value

Reply:
That would make sense to me.